### ISSUE TITLE
No-unused-components reporting incorrectly

### ISSUE THREAD

**Issue by half2me** [vuejs/eslint-plugin-vue#571]
I use the following component: https://github.com/crowdbotics/v-img

I have this:
```html
<template>
  <img v-img src="">
</template>
```
```javascript
import VueImg from 'v-img';
...
components: {
  ...
  VueImg,
},
```

When I run `eslint` I get the following error:
`error  The "VueImg" component has been registered but not used  vue/no-unused-components`

I'm guessing this is probably because the component is not instantiated directly, but rather just attached to the `<img>` tag.

I'm using `^5.0.0-beta.3`

**Comment by half2me** [comment]
possible duplicate of #556

**Comment by michalsnik** [comment]
According to this plugins' documentation you don't have to register any component, but only the plugin and let it do the job (it will register `v-img` directive for you). `VueImg` is a plugin, not component, so you do it like this:
```js
import Vue from 'vue';
import VueImg from 'v-img';

Vue.use(VueImg);
```

**Comment by half2me** [comment]
@michalsnik but if I don't want to register it globally, but just for a single component?

**Comment by alex-shamshurin** [comment]
I still have the same issue

**Comment by half2me** [comment]
It should probably be put in `directives: {VueImg}` instead

**Comment by trajche18** [comment]
Hi @michalsnik  and others, have you found a solution regarding this issue? I am experiencing the same. 

`import VButton from '@/components/common/VButton'`
`import VCombobox from '@/components/common/VCombobox'`
`import VCheckbox from '@/components/common/VCheckbox'`

` components: {
    VButton,
    VCombobox,
    VCheckbox
  }`

The first two components get recognized with no problem. When it comes to VCheckbox, I get: component registered, but not used?

Looking forward to hearing from you. Cheers

**Comment by armano2** [comment]
@trajche18 Are you using it in template?

**Comment by trajche18** [comment]
> @trajche18 Are you using it in template?

Hi @armano2, I certainly am.

`import VCheckbox from '@/components/common/VCheckbox.vue'`

`  .add('ARIA Checkbox', () => ({`
    `components: { VCheckbox },`
  `  template:`
    `  '<v-stats-card :items="items" label="I agree to the Terms and Conditions" ' +`
    `  ':rules="[v => !!v || \'You must agree with the terms and conditions in order to continue.\']" ' +`
    `  'arialabel="You must agree with the terms." ' +`
   `   '@click="onClicked(\'checkbox clicked\')" />',`
   ` methods`
`  }))`

**Comment by trajche18** [comment]
Any updates on this @armano2 and other collaborators?
